### ISSUE TITLE
bug: regression: pressing Esc while previewing an image does not dismiss it

### ISSUE THREAD

**Issue by infi** [revoltchat/revite#686]
### What happened?

Previewing an image and subsequently pressing Esc should dismiss the preview as it has before.

Instead, nothing happens anymore.

![image](https://user-images.githubusercontent.com/45950794/173694617-81da5188-b752-4366-8bf4-25f26dbe4a49.png)

### Branch

Nightly (nightly.revolt.chat)

### Commit hash

a190a51d0b597370e66de96f7c94592debb7705c

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### Desktop

- [ ] Yes, this bug is specific to Revolt Desktop and is *not* an issue with Revolt Desktop itself.

### PWA

- [ ] Yes, this bug is specific to the PWA.

**Comment by LedaThemis** [comment]
this is because the ImageViewer is now using the UI library
https://github.com/revoltchat/revite/blob/a190a51d0b597370e66de96f7c94592debb7705c/src/context/intermediate/popovers/ImageViewer.tsx#L6

the old modal component had the following (which listened for "Escape" key clicks):
```js
useEffect(() => {
    if (props.disallowClosing) return;

    function keyDown(e: KeyboardEvent) {
        if (e.key === "Escape") {
            onClose();
        }
    }

    document.body.addEventListener("keydown", keyDown);
    return () => document.body.removeEventListener("keydown", keyDown);
}, [props.disallowClosing, onClose]);
```
changed in https://github.com/revoltchat/revite/commit/41e533ab59ef2f32554632e417ffff6b1ad31cca

**Comment by LedaThemis** [comment]
Fixed in 0d3f29515e3ec8f1aab179914aa6634d323b2ded, [ImageViewer](https://github.com/revoltchat/revite/blob/f79288826849d0a6e64cb4e1e85b2830ccecc47b/src/controllers/modals/components/ImageViewer.tsx) now uses [ModalController](https://github.com/revoltchat/revite/blob/f79288826849d0a6e64cb4e1e85b2830ccecc47b/src/controllers/modals/ModalController.tsx) which depends on [ModalRenderer](https://github.com/revoltchat/revite/blob/f79288826849d0a6e64cb4e1e85b2830ccecc47b/src/controllers/modals/ModalRenderer.tsx#L13-L14).

**Comment by insertish** [comment]
Closing due to rewrite, marking as potential issue in future by linking to https://github.com/revoltchat/frontend/issues/134.
